contracts/multisig/src/types.rs

```diff
@@ -2,31 +2,85 @@ use std::{collections::HashMap, fmt};
 
 use axelar_wasm_std::Snapshot;
 use cosmwasm_schema::cw_serde;
-use cosmwasm_std::{from_binary, Addr, HexBinary, StdResult};
+use cosmwasm_std::{from_binary, Addr, HexBinary, StdError, StdResult};
 use cw_storage_plus::{KeyDeserialize, PrimaryKey};
 
-use crate::ContractError;
+use crate::{secp256k1::ecdsa_verify, ContractError};
 
 pub trait VerifiableSignature {
     fn verify(&self, msg: &MsgToSign, pub_key: &PublicKey) -> Result<bool, ContractError>;
 }
 
 #[cw_serde]
-pub struct PublicKey(HexBinary);
+pub enum KeyType {
+    ECDSA,
```

Comment:
I noticed that the JSON representation of this looks like `"key_type":"e_c_d_s_a"`. I suggest we use serde rename macro (https://serde.rs/variant-attrs.html#rename) or rename_all so it doesn't have underscores

Reply:
Good catch @eguajardo. First, it should be `Ecdsa` instead of `ECDSA`. Second, should add `#[serde(tag = "type")]` to the enum.